capstone/src/main/webapp/scripts/poll.js

```diff
@@ -0,0 +1,137 @@
+const BAR_WIDTH = "690";
+const BAR_HEIGHT = "55";
+
+function getPollOptions() {
+  fetch("/poll")
+    .then(response => response.json())
+    .then(options => {
+      options.forEach(option => {
+        createOptionElementLoad(option);
+      });
+    });
+}
+
+function createOptionElementLoad(option) {
+  let innerBar = "200";
+  let numVotes = option["userVotes"].length;
+  let text = option["text"];
+
+  const optionsContainer = document.getElementById("options-container");
+  const optionContainer = document.createElement("option-container");
+  optionContainer.setAttribute("class", "option-container");
+  //Add challenge name
+  const challengeName = document.createElement("p");
+  const challengeText = document.createTextNode(text);
+  challengeName.appendChild(challengeText);
+  optionContainer.appendChild(challengeName);
+  //Poll bar
+  const svgElement = document.createElementNS(
+    "http://www.w3.org/2000/svg",
+    "svg"
+  );
+  svgElement.setAttribute("width", BAR_WIDTH);
+  svgElement.setAttribute("height", BAR_HEIGHT);
+  const outerRect = document.createElementNS(
+    "http://www.w3.org/2000/svg",
+    "rect"
+  );
+  outerRect.setAttribute("width", BAR_WIDTH);
+  outerRect.setAttribute("height", BAR_HEIGHT);
+  outerRect.setAttribute("class", "poll-rect");
+  const innerRect = document.createElementNS(
+    "http://www.w3.org/2000/svg",
+    "rect"
+  );
+  innerRect.setAttribute("width", innerBar);
+  innerRect.setAttribute("height", BAR_HEIGHT);
+  innerRect.setAttribute("class", "poll-inner-rect");
+  svgElement.appendChild(outerRect);
+  svgElement.appendChild(innerRect);
+  optionContainer.appendChild(svgElement);
+  //Create checkbox
+  const checkContainer = document.createElement("span");
+  checkContainer.setAttribute("class", "check-container");
+  const inputElement = document.createElement("input");
+  inputElement.setAttribute("type", "checkbox");
+  inputElement.setAttribute("id", text);
+  const labelElement = document.createElement("label");
+  labelElement.htmlFor = text;
+  checkContainer.appendChild(inputElement);
+  checkContainer.appendChild(labelElement);
+  optionContainer.appendChild(checkContainer);
+  //Create votes label
+  const votesLabel = document.createElement("div");
+  votesLabel.setAttribute("class", "votes-label");
+  const votesString = numVotes === 1 ? "task" : "tasks";
+  votesLabel.innerText = `${numVotes} ${votesString}`;
+  optionContainer.appendChild(votesLabel);
+  //Append each option to outer options container
+  optionsContainer.appendChild(optionContainer);
+}
+
+function createOptionElement(text) {
```

Comment:
This is where a templating language can help out. IDK what you're plan is, but maybe consider the following (maybe not ALL within this CL):
- Try moving all styles into your .css.
- Add classnames into this logic to access your styles.
- Consider building this dom structure as a string where you can inject the values you care about. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals 
```javascript
let person = 'Mike';
let age = 28;

function myTag(strings, personExp, ageExp) {
  let str0 = strings[0]; // "That "
  let str1 = strings[1]; // " is a "

  // There is technically a string after
  // the final expression (in our example),
  // but it is empty (""), so disregard.
  // let str2 = strings[2];

  let ageStr;
  if (ageExp > 99){
    ageStr = 'centenarian';
  } else {
    ageStr = 'youngster';
  }

  // We can even return a string built using a template literal
  return `${str0}${personExp}${str1}${ageStr}`;
}

let output = myTag`That ${ person } is a ${ age }`;

console.log(output);
```
- Consider moving this into an HTML file and seeing if you can do the same thing.